capstone/src/main/java/com/google/sps/servlets/UpdateAssociationServlet.java

```diff
@@ -55,17 +80,26 @@ public void destroy() {
    *
    * @return an arraylist of the comment text
    */
-  private ArrayList<String> getComments() {
+  private ArrayList<AssociationInput> getComments() {
     Query query = new Query(SURVEY_ENTITY_KIND);
     PreparedQuery results = datastore.prepare(query);
 
-    ArrayList<String> comments = new ArrayList<String>();
+    ArrayList<AssociationInput> comments = new ArrayList<AssociationInput>();
     for (Entity e : results.asIterable()) {
       if (e.getProperty("association-processed") == null
           || !((boolean) e.getProperty("association-processed"))) {
-        comments.add((String) e.getProperty(COMMENT_PROPERTY));
-        e.setProperty("association-processed", true);
-        datastore.put(e);
+        String message = (String) e.getProperty(COMMENT_PROPERTY);
+        try {
+          ArrayList<String> scope =
+              (ArrayList<String>)
+                  (new MapData()).getPrecincts((String) e.getProperty(ZIPCODE)).clone();
+          scope.add("SF");
+          comments.add(new AssociationInput(message, scope));
+          e.setProperty("association-processed", true);
+          datastore.put(e);
+        } catch (NullPointerException exception) {
+          System.err.println("Invalid zip code in response");
```

Comment:
Should you also log out the zip code value that is invalid here to make it easier to debug?